tempodb/encoding/vparquet4/schema.go

```diff
@@ -0,0 +1,920 @@
+				// later. set all to 0
+				ss.NestedSetLeft = 0
+				ss.NestedSetRight = 0
+				ss.ParentID = 0
+
+				ss.SpanID = s.SpanId
+				ss.ParentSpanID = s.ParentSpanId
+				ss.Name = s.Name
+				ss.Kind = int(s.Kind)
+				ss.TraceState = s.TraceState
+				if s.Status != nil {
+					ss.StatusCode = int(s.Status.Code)
+					ss.StatusMessage = s.Status.Message
+				} else {
+					ss.StatusCode = 0
+					ss.StatusMessage = ""
+				}
+				ss.StartTimeUnixNano = s.StartTimeUnixNano
+				ss.DurationNano = s.EndTimeUnixNano - s.StartTimeUnixNano
+				ss.DroppedAttributesCount = int32(s.DroppedAttributesCount)
+				ss.DroppedEventsCount = int32(s.DroppedEventsCount)
+				ss.HttpMethod = nil
+				ss.HttpUrl = nil
+				ss.HttpStatusCode = nil
+				ss.DedicatedAttributes = DedicatedAttributes{}
+
+				ss.Links = extendReuseSlice(len(s.Links), ss.Links)
+				for ie, e := range s.Links {
+					linkToParquet(e, &ss.Links[ie])
+				}
+
+				ss.DroppedLinksCount = int32(s.DroppedLinksCount)
+
+				ss.Attrs = extendReuseSlice(len(s.Attributes), ss.Attrs)
+				attrCount := 0
+				for _, a := range s.Attributes {
+					written := false
+
+					switch a.Key {
+					case LabelHTTPMethod:
+						strVal, ok := a.Value.Value.(*v1.AnyValue_StringValue)
+						if ok {
+							ss.HttpMethod = &strVal.StringValue
+							written = true
+						}
+					case LabelHTTPUrl:
+						strVal, ok := a.Value.Value.(*v1.AnyValue_StringValue)
+						if ok {
+							ss.HttpUrl = &strVal.StringValue
+							written = true
+						}
+					case LabelHTTPStatusCode:
+						intVal, ok := a.Value.Value.(*v1.AnyValue_IntValue)
+						if ok {
+							ss.HttpStatusCode = &intVal.IntValue
+							written = true
+						}
+					}
+
+					if !written {
+						// Dynamically assigned dedicated span attribute columns
+						if spareColumn, exists := dedicatedSpanAttributes.get(a.Key); exists {
+							written = spareColumn.writeValue(&ss.DedicatedAttributes, a.Value)
+						}
+					}
+
+					if !written {
+						// Other attributes put in generic columns
+						attrToParquet(a, &ss.Attrs[attrCount])
+						attrCount++
+					}
+				}
+				ss.Attrs = ss.Attrs[:attrCount]
+			}
+		}
+	}
+
+	ot.StartTimeUnixNano = traceStart
+	ot.EndTimeUnixNano = traceEnd
+	ot.DurationNano = traceEnd - traceStart
+	ot.RootSpanName = ""
+	ot.RootServiceName = ""
+
+	if rootSpan != nil && rootBatch != nil && rootBatch.Resource != nil {
+		ot.RootSpanName = rootSpan.Name
+
+		for _, a := range rootBatch.Resource.Attributes {
+			if a.Key == LabelServiceName {
+				ot.RootServiceName = a.Value.GetStringValue()
+				break
+			}
+		}
+	}
+
+	return ot, assignNestedSetModelBoundsAndServiceStats(ot)
+}
+
+func instrumentationScopeToParquet(s *v1.InstrumentationScope, ss *InstrumentationScope) {
+	if s == nil {
+		ss.Name = ""
+		ss.Version = ""
+	} else {
+		ss.Name = s.Name
+		ss.Version = s.Version
+	}
+
+	// TODO: handle attributes correctly once they are added to the proto
```

Comment:
these have been added to the proto. i don't think we should handle that in this PR, but perhaps make an issue?

https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/common/v1/common.proto#L79-L80

Reply:
I'm aware of this and have it on my post merge TODO list :+1: